src/future.rs

```diff
@@ -1,3 +1,5 @@
+#![allow(dead_code)]
```

Comment:
Intentionally ignore unused code lints in this module. I am going to address this in a follow up PR, there are even some dependencies that are no longer needed.